client.go

```diff
@@ -161,7 +178,31 @@ func (c *StatsdClient) send(stat string, format string, value interface{}) error
 	}
 	stat = strings.Replace(stat, "%HOST%", Hostname, 1)
 	format = fmt.Sprintf("%s%s:%s", c.prefix, stat, format)
-	_, err := fmt.Fprintf(c.conn, format, value)
+	metricValue := fmt.Sprintf(format, value)
+
+	var err error
+
+	if c.flushInterval != 0 {
+		if !c.timerRunning {
+			c.startTimer()
+		}
+
+		c.statsMutex.Lock()
+
+		if c.buffer.Len() + len(format) > c.maxPacketSize {
+			c.flushStatBuffer()
+			c.Timer.Reset(c.flushInterval)
+		}
+
+		if c.buffer.Len() == 0 {
+			c.buffer.Write([]byte(metricValue))
+		} else {
+			fmt.Fprintf(&c.buffer, "\n%s", metricValue)
```

Comment:
`c.buffer.Write` should be really effective... 

can we re-write it into:
    if c.buffer.Len() > 0 {
      c.buffer.WriteByte('\n')
    }

    c.buffer.WriteString(metricValue)